### PR TITLE
GraphQL for compute node graph

### PR DESCRIPTION
**Pull request by schrockn**
This is a proposed schema for accessing the "compute node graph". Given a config and a pipeline, we produce a logical execution plan of pure lambdas consisting of transform, expectations, and other types of compute. This is the "compiler" layer of the system that is the basis of a physical execution plan on various workflow engines.

In the short term we can use this as an add-on to the config editor to visualize in more specificity what the computation will be as a result of a particular config.

I also suspect we will end up using this for more advanced debugging and tracing tooling in the future.

Feel free to give feedback on anything, including naming. I don't really like the "compute node" verbiage. It was pretty arbitrary and I came up with it a long time ago, so I'm willing to do a system-wide name change if something else pops up.

### CODE REVIEW

File: python_modules/dagit/dagit/schema.py [hunk 1]
```diff
@@ -33,19 +37,140 @@ def results_or_errors(root_obj, fn, info, *argv):
...
+    except Exception:  # pylint: disable=broad-except
         return [PythonError(*sys.exc_info())]
 
 
+def non_null_list(ttype):
+    return graphene.NonNull(graphene.List(graphene.NonNull(ttype)))
+
+
+class ComputeNodeGraph(graphene.ObjectType):
+    def __init__(self, compute_node_graph):
```
Comment (freiksenet) [inline python_modules/dagit/dagit/schema.py]: We've been following Django convention for ordering of stuff in GraphQL Schema, so fields before the `__init__`, `Meta` before the init, methods after `__init__`.

File: python_modules/dagit/dagit/schema.py [hunk 2]
```diff
@@ -33,19 +37,140 @@ def results_or_errors(root_obj, fn, info, *argv):
...
-    except:
+    except Exception:  # pylint: disable=broad-except
         return [PythonError(*sys.exc_info())]
 
 
+def non_null_list(ttype):
+    return graphene.NonNull(graphene.List(graphene.NonNull(ttype)))
+
+
+class ComputeNodeGraph(graphene.ObjectType):
```
Comment (freiksenet) [inline python_modules/dagit/dagit/schema.py]: Probably needs at least a backreference to pipeline, maybe in future to config that was used to create it.

File: python_modules/dagit/dagit/schema.py [hunk 3]
```diff
@@ -93,14 +218,23 @@ def resolve_types(self, info, pipelineName):
...
     description = graphene.String()
-    solids = graphene.NonNull(graphene.List(lambda: graphene.NonNull(Solid)))
-    contexts = graphene.NonNull(graphene.List(lambda: graphene.NonNull(PipelineContext)))
+    solids = non_null_list(lambda: Solid)
+    contexts = non_null_list(lambda: PipelineContext)
     environment_type = graphene.NonNull(lambda: Type)
+    computeNodeGraph = graphene.Field(
+        graphene.NonNull(lambda: ComputeNodeGraph),
+        args={
+            'config': graphene.Argument(graphene.NonNull(graphene.String)),
```
Comment (freiksenet) [inline python_modules/dagit/dagit/schema.py]: I've been thinking about that after our call and I'm not 100% sure config should be passed as string. However it's fine for this moment.

Should this be optional though? Is empty config a valid config?

File: python_modules/dagit/dagit_tests/test_graphql.py [hunk 4]
```diff
@@ -137,296 +141,115 @@ def test_pipelines():
...
+        }
     )
 
-    if result.errors:
-        raise Exception(result.errors)
-
     assert result.data
     assert not result.errors
 
+    compute_node_graph_data = result.data['pipeline']['computeNodeGraph']
```
Comment (freiksenet) [inline python_modules/dagit/dagit_tests/test_graphql.py]: Should we snapshot test here instead?

File: python_modules/dagster/dagster/core/execution.py [hunk 5]
```diff
@@ -258,7 +260,25 @@ def _validate_environment(environment, pipeline):
             )
 
 
-def _create_config_value(config_type, config_input):
+def create_compute_node_graph(pipeline, yaml_config):
```
Comment (freiksenet) [inline python_modules/dagster/dagster/core/execution.py]: This function is what makes me unhappy about passing strings in GraphQL :( Tbh, I'd rather have yaml be implementation detail, not something that ingrained into our system.

File: python_modules/dagit/dagit/schema.py (hunk 1 above)
Comment (schrockn) [reply python_modules/dagit/dagit/schema.py]: ✔️

File: python_modules/dagit/dagit/schema.py (hunk 2 above)
Comment (schrockn) [reply python_modules/dagit/dagit/schema.py]: ✔️

File: python_modules/dagit/dagit_tests/test_graphql.py (hunk 4 above)
Comment (schrockn) [reply python_modules/dagit/dagit_tests/test_graphql.py]: ya can do that. at first i generally develop tests incrementally and want super granular errors.

File: python_modules/dagster/dagster/core/execution.py (hunk 5 above)
Comment (schrockn) [reply python_modules/dagster/dagster/core/execution.py]: per discussion on slack, going to have this be the JSON/GenericScalar/InputAnyType

File: python_modules/dagster/dagster/core/execution.py [hunk 6]
```diff
@@ -18,6 +18,7 @@
 from contextlib import contextmanager
 import json
 import itertools
+import yaml
```
Comment (freiksenet) [inline python_modules/dagster/dagster/core/execution.py]: [nit] unused

**Pull request merged**